### PR TITLE
Update test documentation with circleCI e2e_dind details

### PR DESCRIPTION
**Pull request by tiswanso**
Attempt to explain how e2e_dind works

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -1,22 +1,119 @@
...
+
+#### CircleCI e2e Tests
+
+The Istio CNI e2e_dind CircleCI job is patterned on the KDC CircleCI jobs.  The
+e2e_dind job creates a 3-node Kubernetes cluster via docker-in-docker (DinD)
+using the [kubeadm-dind-cluster (KDC)](https://github.com/kubernetes-sigs/kubeadm-dind-cluster)
+project.  Each node in the Kubernetes cluster is actually a docker container
+hosted on the CircleCI job's "remote" docker instance.  In order to run the
+Istio e2e_simple tests against that environment, a e2e_runner container is
+run within the same network namespace as the kubernetes master node
```
Comment (john-a-joyce) [inline test/README.md]: kubernetes should be capitalized this is in multiple places and you should probably search for an istio vs. Istio

File: test/README.md [hunk 2]
```diff
@@ -1,22 +1,119 @@
...
+
+1. The e2e_dind job runs the e2e_runner container on the same docker network as the
+   kubernetes cluster's kube-master node.
+
+   1. The e2e_simple test is run by the e2e_runner exec script.
+
+##### kubeadm-dind-cluster (KDC) Comments
+
+- Implementations for multiple Kubernetes versions--since 1.8
+- Supports multiple CNI plugin types--weave, flannel, calico, calico_kdd, kube_router
```
Comment (john-a-joyce) [inline test/README.md]: Not 100% sure, but I think most of these need to be capitalized and there are multiple instances through out the file

File: test/README.md [hunk 3]
```diff
@@ -1,22 +1,119 @@
...
-This page describes the testing strategy for the CNI plugin and how to run the tests.
+This page describes the testing strategy for the Istio CNI plugin and how to run the tests.
 
 ## Types of tests
 
-The CNI will have the following tests available.
+The CNI has the following tests available.
 
-1. Unit tests
+1. Unit, Functional, and lint tests
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply test/README.md]: done

File: test/README.md (hunk 2 above)
Comment (tiswanso) [reply test/README.md]: I'm holding off on this as these are type values in the KDC config and this doc is developer/tester focused so don't want to get too pedantic here.

File: test/README.md [hunk 4]
```diff
@@ -1,22 +1,119 @@
...
+
+### Istio Prow (GKE)
+
+PR triggers from the istio/istio repo will execute a new prow job added to istio/test-infra called
+e2e_simple_cni.  The test wrapper script is added via PR [9557](https://github.com/istio/istio/pull/9577)
+
+The "mason" component of istio/test-infra has been modified to create GKE clusters with
+the [network policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy)
+feature enabled which is a requirement for GKE to enable the CNI network-plugin
+in the installed kubelets.  This has already been merged into istio/test-infra.
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply test/README.md]: done

File: README.md [hunk 5]
```diff
@@ -215,25 +215,9 @@ To make use of the `istio-cni` chart from another chart:
...
-```
-
-#### Current Tests
-1. install-test (`make install-test`)
-   1. Requires docker
-   1. docker runs the `install-cni` container with test dirs mounted and env vars set to known Kubernetes settings
-   1. does file compares to determine if the results match the expected configuration
+The Istio CNI testing strategy and execution details are explained [here](test/README.md).
 
 ## Implementation Details
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: I removed the TODOs.  Is that what you're recommending or the whole section?  If the whole section, I can reorg that in another PR.

[the remark this answers is not in the document]
Comment (john-a-joyce) [reply README.md]: Mostly the TODOs was the main thing

File: test/README.md (hunk 2 above)
Comment (john-a-joyce) [reply test/README.md]: If they are types then that is fine,  but you run amok of copyright/trademark if you don't properly capitalize things that should be capitalized.

**Pull request merged**